jraft-core/src/main/java/com/alipay/sofa/jraft/Node.java

```diff
@@ -226,4 +227,32 @@
      * @throws LogIndexOutOfBoundsException  the special index is out of bounds.
      */
     UserLog readCommittedUserLog(final long index);
+
+    /**
+     * SOFAJRaft End User can implement the ReplicatorStateListener interface by themselives
+     * So users can do their own logical operator in this listener when replicator started ,stopped or has some errors
+     *
+     * @param replicatorStateListener register ReplicatorStateListener which is implemented by users
+     */
+    void registerReplicatorStateListener(final Replicator.ReplicatorStateListener replicatorStateListener);
+
+    /**
+     * End User can remove their implement the ReplicatorStateListener interface by themselives
+     *
+     * @param replicatorStateListener need to remove the ReplicatorStateListener which has been registed by users
+     */
+    void removeReplicatorStateListener(final Replicator.ReplicatorStateListener replicatorStateListener);
+
+    /**
+     * Remove all the ReplicatorStateListeners which have been registed by users
+     *
+     */
+    void clearReplicatorStateListeners();
+
+    /**
+     * Get the ReplicatorStateListener which have been registed by users.
```

Comment:
registered?

Reply:
Get the replicatorStateListeners

Reply:
Okay,I will fix this codes comments and then push it again.Thanks!